specs/core/0_beacon-chain.md

```diff
@@ -541,7 +541,7 @@ class BeaconState(Container):
 #### `integer_squareroot`
 
 ```python
-def integer_squareroot(n: uint64) -> uint64:
+def integer_squareroot(n: int) -> int:
```

Comment:
Using `uint64` allows us to remove the untriggerable `assert n >= 0`.

Reply:
It is a generic helper that has nothing to do with the SSZ types and works on all non-negative ints. Similar to `int_to_bytes` in typing.

I think we add in a comment about the value expected to be non-negative if we don't want th assert in there.

Reply:
I'm inclined to signal to implementers that we don't use unsigned integers anywhere. It's a nice conceptual simplification. Tempted to change the various `int` to `uint64`.

Reply:
reasonable. handling it

Reply:
You could make it `uint` if you want to. Just include it in the spec builder in imports. But prefer not to break anything like that, with the amount of time for freeze. We can do such super minor presentation things post-freeze.

Reply:
The only place I can think of having `uint64`s fail is for BLS privkeys. Are there any other places you can think of @protolambda? I am not saying I don't favour the conservative route too.

Reply:
went with `uint64`. It's our default int type and fine for these purposes.
We can change after if nedded as it is almost entirely cosmetic